kernel/src/transaction.rs

```diff
@@ -240,6 +240,11 @@ mod tests {
 
         // FIXME actual assertions
         assert_eq!(actions_schema.fields().collect::<Vec<_>>().len(), 6);
+        let DataType::Struct(struct_type) = actions_schema.field("commitInfo").unwrap().data_type()
```

Comment:
```rust
assert!(matches!(actions_schema.field("commitInfo").map(|f| f.data_type()), Some(DataType::Struct(_))
```

Reply:
hm but i need to pattern match to get out the struct type and check it's got 4 fields

Reply:
Oh, I see it looked like you were just trying to see if it was a struct.

Reply:
i'll probably just try to improve this whole chunk anyways :)